gdnative-core/src/core_types/transform2d.rs

```diff
@@ -8,3 +8,188 @@ pub struct Transform2D {
+    pub fn from_axis_origin(x_axis: Vector2, y_axis: Vector2, origin: Vector2) -> Transform2D {
+        Transform2D {
+            x: x_axis,
+            y: y_axis,
+            origin,
+        }
+    }
+
+    /// Constructs the transform from a given angle (in radians) and position.
+    pub fn from_rotation_position(rotation: f32, position: Vector2) -> Transform2D {
+        let cr = f32::cos(rotation);
+        let sr = f32::sin(rotation);
+
+        Transform2D {
+            x: Vector2::new(cr, sr),
+            y: Vector2::new(-sr, cr),
+            origin: position,
+        }
+    }
+
+    /// Returns the inverse of the transform, under the assumption that the transformation is composed of rotation, scaling and translation.
+    pub fn affine_inverse(&self) -> Transform2D {
+        let mut inverted = *self;
+
+        let det = self.basis_determinant();
+        // assert!(det != 0.0, "The determinant cannot be zero");
+        let idet = 1.0 / det;
+
+        std::mem::swap(&mut inverted.x.x, &mut inverted.y.y);
+        inverted.x *= Vector2::new(idet, -idet);
+        inverted.y *= Vector2::new(-idet, idet);
+        inverted.origin = inverted.basis_xform(-inverted.origin);
+
+        inverted
+    }
+
+    /// Returns a vector transformed (multiplied) by the basis matrix.
+    ///
+    /// This method does not account for translation (the origin vector).
+    pub fn basis_xform(&self, v: Vector2) -> Vector2 {
+        Vector2::new(self.x.dot(v), self.y.dot(v))
+    }
+
+    /// Returns a vector transformed (multiplied) by the inverse basis matrix.
+    ///
+    /// This method does not account for translation (the origin vector).
+    pub fn basis_xform_inv(&self, v: Vector2) -> Vector2 {
+        Vector2::new(self.tdotx(v), self.tdoty(v))
+    }
+
+    /// Transforms the given Vector2, Rect2, or PoolVector2Array by this transform.
+    pub fn xform(&self, v: Vector2) -> Vector2 {
+        Vector2::new(self.tdotx(v), self.tdoty(v)) + self.origin
+    }
+
+    /// Inverse-transforms the given Vector2, Rect2, or PoolVector2Array by this transform.
+    pub fn xform_inv(&self, v: Vector2) -> Vector2 {
+        let v = v - self.origin;
+        Vector2::new(self.x.dot(v), self.y.dot(v))
+    }
+
+    /// Returns the transform's origin (translation).
+    pub fn get_origin(&self) -> Vector2 {
+        self.origin
+    }
+
+    /// Sets the transform's origin (translation).
+    pub fn set_origin(&mut self, origin: Vector2) {
+        self.origin = origin;
+    }
+
+    /// Translates the transform by the given offset, relative to the transform's basis vectors.
+    ///
+    /// Unlike rotated() and scaled(), this does not use matrix multiplication.
+    pub fn translated(&self, origin: Vector2) -> Transform2D {
+        let mut new = *self;
+        new.set_origin(origin);
+        new
+    }
+
+    /// Returns the transform's rotation (in radians).
+    pub fn get_rotation(&self) -> f32 {
```

Comment:
Getters should not have a `get_` prefix according to current convention. For `scale()` it can be a bit confusing, but we would need to change this everywhere.